src/sqlfluff/core/parser/segments/base.py

```diff
@@ -885,15 +885,14 @@ def select_children(
                 buff.append(seg)
         return buff
 
-    def recursive_crawl(self, *seg_type, recurse_into=True):
-        """Recursively crawl for segments of a given type.
-
-        Args:
-            seg_type: :obj:`str`: one or more type of segment
-                to look for.
-            recurse_into: :obj:`bool`: When an element of type "seg_type" is
-                found, whether to recurse into it.
-        """
+    def recursive_crawl_all(self):
+        """Recursively crawl all descendant segments."""
+        yield self
+        for seg in self.segments:
+            yield from seg.recursive_crawl_all()
+
+    def recursive_crawl(self, *seg_type: str, recurse_into: bool = True):
+        """Recursively crawl for segments of a given type."""
```

Comment:
Did you miss this one @barrywhart ?

```suggestion
        """Recursively crawl for segments of a given type.
        Args:
            seg_type: :obj:`str`: one or more type of segment
                to look for.
            recurse_into: :obj:`bool`: When an element of type "seg_type" is
                found, whether to recurse into it.
        """
```